src/main/java/org/jabref/gui/actions/JabRefAction.java

```diff
@@ -23,11 +24,23 @@ public JabRefAction(Action action, KeyBindingRepository keyBindingRepository) {
     }
 
     public JabRefAction(Action action, Command command, KeyBindingRepository keyBindingRepository) {
+        this(action, command, keyBindingRepository, null);
+    }
+
+    /**
+     * especially for the track execute when the action run the same function but from different source.
+     * @param source is a string contains the source, for example "button"
+     */
+    public JabRefAction(Action action, Command command, KeyBindingRepository keyBindingRepository, String source) {
         this(action, keyBindingRepository);
 
         setEventHandler(event -> {
             command.execute();
-            trackExecute(getActionName(action, command));
+            if (StringUtil.isNullOrEmpty(source)) {
+                trackExecute(getActionName(action, command));
+            } else {
+                trackExecute(getActionName(action, command) + "From" + source);
```

Comment:
The ApplicationInsights library that we use supports a special way to submit additional details (which I completely forgot until 5 mins ago): https://docs.microsoft.com/en-us/azure/azure-monitor/app/api-custom-events-metrics#properties
Could you please send the `source` as a property please.
https://github.com/JabRef/jabref/blob/7561724b191e890d8687eeb863bf8f0a59a516e0/src/main/java/org/jabref/gui/JabRefFrame.java#L1016-L1022